### DISCUSSION TITLE
version from project.toml not used in GitHub build action

### DISCUSSION THREAD

**Discussion by davidvlaminck** [pypa/setuptools#3812] [Q&A] [answered]
Hi,
I've been stomped by this issue and need some help.
I've made a GitHub action to make a release to PyPI. The release flow itself has worked before but for some reason it no longer does.

My [project.toml](https://github.com/davidvlaminck/OTLMOW-Model/blob/master/pyproject.toml) has a fixed version number. I don't use a setup.cfg or setup.py because everything I need is in the toml file.

When I run the build command "python3 -m build" locally, in my venv, it creates a wheel and a tar for this version (2.6.3 in this case)
When I run the GitHub action for the [release](https://github.com/davidvlaminck/OTLMOW-Model/actions/runs/4126144975/workflow), that uses the command `python3 -m pip install --upgrade build && python3 -m build` it doesn't work correctly and still uses the previous version (2.6.2) for some reason. I end up with a 2.6.2 tar/wheel which fails to upload because it already exists.

Does anyone have an idea what causes this problem and how to fix it?

**Comment by jwodder** [comment, marked as answer]
GitHub Actions says that [this failed run](https://github.com/davidvlaminck/OTLMOW-Model/actions/runs/4126144975) was against tag `v2.6.3`, which is [this tree](https://github.com/davidvlaminck/OTLMOW-Model/tree/41db70f3ff954859abf0c3fce4261692d2804bed), but [the `pyproject.toml` in that commit](https://github.com/davidvlaminck/OTLMOW-Model/blob/41db70f3ff954859abf0c3fce4261692d2804bed/pyproject.toml) still says `version = '2.6.2'`.  Apparently, you created the tag before updating the version.

**Reply by davidvlaminck** [reply]
Ah, so, it reruns on the same tree. I didn't realize that! When I ran it manually, it used the latest tree, and it build like expected. Thanks for pointing me in the right direction!
